clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -191,9 +191,7 @@ private void createKeystore(String type, String path, Password password, Passwor
     private void createTruststore(String type, String path, Password password) {
         if (path == null && password != null) {
             throw new KafkaException("SSL trust store is not specified, but trust store password is specified.");
-        } else if (path != null && password == null) {
-            throw new KafkaException("SSL trust store is specified, but trust store password is not specified.");
-        } else if (path != null && password != null) {
+        } else if (path != null) {
             this.truststore = new SecurityStore(type, path, password);
```

Comment:
I suppose we could log a warning. Though, it may indicate that something is wrong or risky which isn't necessarily true (see my jira description).